### PR TITLE
Add support to hide columns in the portability HTML report

### PR DESCRIPTION
**Pull request by jeremymeng**
/cc @twsouthwick @conniey

### CODE REVIEW

File: src/Microsoft.Fx.Portability.Reports.Html/Resources/_PortabilityReport.cshtml [hunk 1]
```diff
@@ -144,7 +144,19 @@
         detailColumnHeaders.Add(LocalizedStrings.TargetTypeHeader);
         detailColumnHeaders.AddRange(Model.TargetHeaders);
         detailColumnHeaders.Add(LocalizedStrings.RecommendedChanges);
-        <table>
+
+        <form>
+            Hide columns: 
+            @* It is not allowed to hide the first (Api name) and the last (Recommended Changes) columns. *@
```
Comment (twsouthwick) [inline src/Microsoft.Fx.Portability.Reports.Html/Resources/_PortabilityReport.cshtml]: Does this mean that we don't want them to or it's not possible?

**Comment by jeremymeng** [comment]
Here's what it looks like after the change:
image

File: src/Microsoft.Fx.Portability.Reports.Html/Resources/_Scripts.cshtml [hunk 2]
```diff
@@ -40,4 +40,32 @@
         var currentClass = beginCollapsed[i].getAttribute('class');
         beginCollapsed[i].setAttribute('class', currentClass.concat(' ToggledOff').trim());
     }
+
+    // Toggle the visibility of a column in the portability report detail
+    // table (id='table_portability_details')
+    function show_hide_column(col_no, do_show) {
+
```
Comment (twsouthwick) [inline src/Microsoft.Fx.Portability.Reports.Html/Resources/_Scripts.cshtml]: nit: newline

**Comment by twsouthwick** [comment]
Looks like the change introduces the selection per-assembly. I think it would make more sense to have it for the whole report

File: src/Microsoft.Fx.Portability.Reports.Html/Resources/_PortabilityReport.cshtml (hunk 1 above)
Comment (jeremymeng) [inline src/Microsoft.Fx.Portability.Reports.Html/Resources/_PortabilityReport.cshtml]: We don't allow as it wouldn't make sense to hide the first column.  Maybe allow hiding the last?

**Comment by twsouthwick** [comment]
Can you show an updated screenshot with the changes?

**Comment by jeremymeng** [comment]
updated screenshot:

image

File: src/Microsoft.Fx.Portability.Reports.Html/Resources/_PortabilityReport.cshtml [hunk 3]
```diff
@@ -106,63 +106,78 @@
...
 </table>
 <div id="details">
-    @foreach (var assembly in Model.OrderedAssembliesByIdentity)
-    {
-        // avoid writing blank details tables (happens if a non-.NET assembly is analyzed)
-        bool isDotNETAssembly = false;
-        foreach (var type in Model.MissingTypes)
+        @{ 
+            var detailColumnHeaders = new List<string>
+            ();
```
Comment (conniey) [inline src/Microsoft.Fx.Portability.Reports.Html/Resources/_PortabilityReport.cshtml]: Move it to the line above into `new List<string>();`

**Comment by jeremymeng** [comment]
Thanks!  I add a small bug fix and squashed the changes.

**Comment by jeremymeng** [comment]
Thanks!

**Pull request merged**